src/method/method_query_handler.cpp

```diff
@@ -246,218 +236,59 @@ namespace cubmethod
   }
 
   int
-  query_handler::set_qresult_info (std::vector<query_result_info> &qinfo)
+  query_handler::set_qresult_info (query_result_info &qinfo)
   {
     int error = NO_ERROR;
-    int num_q_result = m_q_result.size();
     OID ins_oid = OID_INITIALIZER;
     DB_OBJECT *ins_obj_p;
 
-    qinfo.resize (num_q_result);
-    for (int i = 0; i < num_q_result; i++)
+    query_result &qresult = m_query_result;
+    DB_QUERY_RESULT *qres = (DB_QUERY_RESULT *) qresult.result;
+    if (qresult.stmt_type == CUBRID_STMT_INSERT && qres != NULL)
```

Comment:
It needs an explanation why stmt_type must be CUBRID_STMT_INSERT.

Reply:
It is for the GET_GENERATED_KEYS request. I've added a comment.